rules/integrations/gcp/credential_access_gcp_kubernetes_secrets_enumeration.toml

```diff
@@ -0,0 +1,41 @@
+[metadata]
+creation_date = "2021/06/06"
+maturity = "production"
+updated_date = "2021/06/06"
+
+[rule]
+author = ["Elastic","Austin Songer"]
+description = """
+Identifies the unauthorized enumeration of kubernetes secrets by a user.
+"""
+from = "now-20m"
+index = ["filebeat-*", "logs-gcp*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "GCP Kubernetes Secrets Enumeration"
+note = """## Config
+
+The GCP Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://cloud.google.com/kubernetes-engine/docs/how-to/audit-logging",
+]
+risk_score = 47
+rule_id = "90b6cf41-801b-4979-9233-e9a234257cd9"
+severity = "medium"
+tags = ["Elastic", "Cloud", "GCP", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+timestamp_override = "event.ingested"
+type = "query"
+
+query = '''
+event.dataset:(googlecloud.audit or gcp.audit) and event.action:io.k8s.core.v*.secrets.get and event.outcome:success
+'''
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
+
```

Comment:
This one seems to be a good fit here: https://attack.mitre.org/techniques/T1552/007/